public/Aryan Suri/Sidebar/sidebar.js

```diff
@@ -0,0 +1,1645 @@
+  if (!isNaN(sourceValue) || sourceValue == 0) {
+    // If we can make a valid floating-point number, put it in the text box and convert!
+    sourceForm.unit_input.value = sourceValue;
+    ConvertFromTo(sourceForm, targetForm);
+  }
+}
+
+function ConvertFromTo(sourceForm, targetForm) {
+  // Converts the contents of the sourceForm input box to the units specified in the targetForm unit menu and puts the result in the targetForm input box.In other words, this is the heart of the whole script...
+  var propIndex;
+  var sourceIndex;
+  var sourceFactor;
+  var targetIndex;
+  var targetFactor;
+  var result;
+
+  // Start by checking which property we are working in...
+  propIndex = document.property_form.the_menu.selectedIndex;
+
+  // Let's determine what unit are we converting FROM (i.e. source) and the factor needed to convert that unit to the base unit.
+  sourceIndex = sourceForm.unit_menu.selectedIndex;
+  sourceFactor = factor[propIndex][sourceIndex];
+
+  // Cool! Let's do the same thing for the target unit - the units we are converting TO:
+  targetIndex = targetForm.unit_menu.selectedIndex;
+  targetFactor = factor[propIndex][targetIndex];
+
+  // Simple, huh? let's do the math: a) convert the source TO the base unit: (The input has been checked by the CalculateUnit function).
+
+  result = sourceForm.unit_input.value;
+  // Handle Temperature increments!
+  if (property[propIndex] == "Temperature") {
+    result = parseFloat(result) + tempIncrement[sourceIndex];
+  }
+  result = result * sourceFactor;
+
+  // not done yet... now, b) use the targetFactor to convert FROM the base unit
+  // to the target unit...
+  result = result / targetFactor;
+  // Again, handle Temperature increments!
+  if (property[propIndex] == "Temperature") {
+    result = parseFloat(result) - tempIncrement[targetIndex];
+  }
+
+  // Ta-da! All that's left is to update the target input box:
+  targetForm.unit_input.value = result;
+}
+
+// This fragment initializes the property dropdown menu using the data defined above in the 'Data Definitions' section
+window.onload = function (e) {
+  FillMenuWithArray(document.property_form.the_menu, property);
+  UpdateUnitMenu(document.property_form.the_menu, document.form_A.unit_menu);
+  UpdateUnitMenu(document.property_form.the_menu, document.form_B.unit_menu)
+}
+
+// Restricting textboxes to accept numbers + navigational keys only
+document.getElementById('numbersonly').addEventListener('keydown', function (e) {
+  var key = e.keyCode ? e.keyCode : e.which;
+
+  if (!([8, 9, 13, 27, 46, 110, 190].indexOf(key) !== -1 ||
+    (key == 65 && (e.ctrlKey || e.metaKey)) || // Select All 
+    (key == 67 && (e.ctrlKey || e.metaKey)) || // Copy
+    (key == 86 && (e.ctrlKey || e.metaKey)) || // Paste
+    (key >= 35 && key <= 40) || // End, Home, Arrows
+    (key >= 48 && key <= 57 && !(e.shiftKey || e.altKey)) || // Numeric Keys
+    (key >= 96 && key <= 105) // Numpad
+      (key == 190) // Numpad
```

Comment:
This piece of code appears to be very odd. Can this be solved using `<input type="number">` or is this addressing some other issue?

Reply:
This auto deletes non-number values entered into converted